contracts/root/staking/CustomSupernetManager.sol

```diff
@@ -129,9 +134,31 @@ contract CustomSupernetManager is ICustomSupernetManager, Ownable2StepUpgradeabl
         return validators[validator_];
     }
 
+    /**
+     *
+     * @inheritdoc ICustomSupernetManager
+     */
+    function premine(uint256 amount) external {
+        if (address(_rootERC20Predicate) == address(0)) {
+            revert Unauthorized("CustomSupernetManager: UNDEFINED_ROOT_ERC20_PREDICATE");
+        }
+
+        IERC20 nativeTokenRoot = IERC20(_rootERC20Predicate.nativeTokenRoot());
+        if (address(nativeTokenRoot) == address(0)) {
+            revert Unauthorized("CustomSupernetManager: UNDEFINED_NATIVE_TOKEN_ROOT");
+        }
+        require(!_genesis.completed(), "CustomSupernetManager: GENESIS_SET_IS_ALREADY_FINALIZED");
+
+        nativeTokenRoot.safeTransferFrom(msg.sender, address(_rootERC20Predicate), amount);
+        _genesis.insert(msg.sender, 0, amount);
+
+        // slither-disable-next-line reentrancy-events
+        emit AccountPremined(msg.sender, amount);
```

Comment:
Well yes, it would mean that we need to extract multiple blocks and search for the `AccountPremined` log, which is not really convenient. Additionally, it would mean that we are retrieving `initialStake` for the genesis accounts differently than the genesis `balance`, but the mindset was to use the same approach and reuse as much code as possible.